library/cmac.c

```diff
@@ -793,6 +781,17 @@ static int cmac_test_subkeys( int verbose,
         if( ( ret = mbedtls_cipher_setkey( &ctx, key, keybits,
                                        MBEDTLS_ENCRYPT ) ) != 0 )
         {
+            /* When CMAC is implemented by an alternative implementation, or
```

Comment:
Well, if possible, I would prefer not being so permissive. That way, we may miss real problems. I would prefer a solution like in aes.c where we have some extra checks when UNSUPPORTED in returned:
```c
/*                                                                      
 * AES-192 is an optional feature that may be unavailable when          
 * there is an alternative underlying implementation i.e. when          
 * MBEDTLS_AES_ALT is defined.                                          
 */                                                                     
if( ret == MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED && keybits == 192 ) 
{                                                                       
    mbedtls_printf( "skipped\n" );                                      
    continue;                                                           
 }   
```
If the only problems you encountered are because of AES-192, I would prefer something like:
```c
if( ret == MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED && cipher_type == MBEDTLS_CIPHER_AES_192_ECB ) 
        {                                                                       
           ...                            
```

Reply:
Our immediate problem is indeed about AES-192, but this could still be relevant for parts from other manufacturers which have e.g. an AES accelerator which only does AES-128. They sadly do exist :/

Reply:
ok, I understand your point that we may have to improve this again at some point in the future but I'd prefer to be as less permissive as possible and thus just filter the AES-192 case as in aes.c.

Reply:
Done.